src/components/form/signInForm/signInForm.js

```diff
@@ -0,0 +1,130 @@
+import { useState } from 'react'
+import { useDispatch } from 'react-redux'
+import { NavLink } from 'react-router-dom'
+import { useFormik } from 'formik'
+import * as Yup from 'yup'
+
+import Title from '../../title'
+import Form from '../formStyles/form'
+import Wrapper from '../formStyles/inputWrapper'
+import Input from '../formStyles/input'
+import ErrorText from '../formStyles/errorText'
+import Svg from '../formStyles/passwordSvg'
+import Button from '../formStyles/button'
+import Link from '../formStyles/link'
+import TextWrapper from '../formStyles/textWrapper'
+import Text from '../formStyles/text'
+
+import routes from '../../../routes/routes'
+import authActions from '../../../redux/auth/authActions'
+
+const SignInForm = () => {
+  const [togglePass, setTogglePass] = useState(false)
+
+  const dispatch = useDispatch()
+
+  const handleToggle = (e) => {
+    if (e.target.id === 'togglePass') {
+      setTogglePass(!togglePass)
+    }
+
+    return
+  }
+
+  const formik = useFormik({
+    initialValues: {
+      email: '',
+      password: '',
+    },
+    validationSchema: Yup.object({
+      email: Yup.string()
+        .email('Invalid email address')
+        .required('Please enter your email address'),
+      password: Yup.string()
+        .required('Please Enter your password')
+        .matches(
+          /^(?=.*[A-Za-z])(?=.*\d)[A-Za-z\d]{8,}$/,
+          'Must contain 8 characters,at least one letter and one number'
+        ),
+    }),
+    onSubmit: () => dispatch(authActions.registerSuccess()),
+  })
+
+  return (
+    <>
+      <Form onSubmit={formik.handleSubmit}>
+        <Title text="Sign In" />
+
+        <div>
```

Comment:
Could be removed.

Reply:
if i remove this div css rules (&:nth-child(n)::before and last-child)  become unsuported.